File: miscellaneous/ltcd_2178.cpp
//LeetCode - problem 2178 - Greedy Algo

    vector<long long> maximumEvenSplit(long long n) {
        if(n%2) 
            return {};

		vector<long long> ans;
        long long i = 2;
        long long crSum=0;
		
        while((crSum+i)<= n)
        {
            ans.push_back(i);
            crSum+=i;
            i+=2;
        }
		
		int sz = ans.size();
        ans[sz-1] += (n-crSum);
        return ans;
    }

File: miscellaneous/ltcd_2178.py
'''
LeetCode - problem 2178 - Greedy algo
'''

    def maximumEvenSplit(self, finalSum: int) -> List[int]:
        arr = []
        if finalSum % 2 == 0:
            a, i = finalSum // 2, 1 
            while i <= a: 
                arr.append(2*i) 
                a -= i 
                i += 1 
            s = sum(arr)
            arr[-1] += finalSum - s
        return arr
